src/kernel.cr

```diff
@@ -39,6 +57,15 @@ STDOUT = IO::FileDescriptor.from_stdio(1)
 #   This is convenient but slower than with `flush_on_newline` set to `false`.
 #   If you need a bit more performance and you don't care about near real-time
 #   output you can do `STDERR.flush_on_newline = false`.
+#
+# On Unix systems, if the file descriptor is a TTY, the runtime duplicates it.
+# So `STDERR.fd` might not be `2`.
+# The reason for this is to enable non-blocking writes for concurrency. Other fibers
+# can run while waiting on IO. The original file descriptor is
+# inherited from the parent process. Setting it to non-blocking mode would
+# reflect back which can cause problems.
+#
+# On Windows, `STDERR` is always blocking.
```

Comment:
It applies to the rest of the text too, btw, but there's no good reason to start new text with such large repetition.

```suggestion
Ditto, `STDERR.fd` might not be `2`.
```